### PR TITLE
Issue#705 deploy org contract

### PR DESCRIPTION
**Pull request by pldespaigne**

### CODE REVIEW

File: apps/backend-functions/src/internals/algolia.ts [hunk 1]
```diff
@@ -10,15 +10,17 @@ const indexOrganizationsBuilder = () => {
 
 export function storeSearchableOrg(orgId: string, name: string): Promise<any> {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/algolia.ts]: Why not using async ? : 
```typescript
export async function ...(): Promise<boolean> {
 ...
 return true;
}
```

File: apps/backend-functions/src/main.ts [hunk 2]
```diff
@@ -177,7 +177,7 @@ export const onOrganizationCreateEvent = onDocumentCreate(
 /**
  * Trigger: when an organization is updated
  */
-export const onOrganizationUpdateEvent = onDocumentUpdate(
+export const onOrganizationUpdateEvent = onOrgDocumentUpdate( // using `onOrgDocument` instead of `onDocument` for an increase timout of 540s
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/main.ts]: name the method  `onOrganizationDocumentUpdate()` or `onOrganizationUpdate`

File: apps/backend-functions/src/orgs.ts [hunk 3]
```diff
@@ -43,7 +49,28 @@ export function onOrganizationUpdate(
 
   // Update algolia's index
   if (before.name !== after.name) {
-    return storeSearchableOrg(orgID, after.name);
+    throw new Error('Organization name cannot be changed !'); // this will require to change the org ENS name, for now we throw to prevent silent bug
+    // const orgID = context.params.orgID;
+    // await storeSearchableOrg(orgID, after.name); // update org name on algolia
+  }
+
+  // console.log(before, '****************************' ,after);
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/orgs.ts]: Remove this comment

File: apps/backend-functions/src/orgs.ts [hunk 4]
```diff
@@ -43,7 +49,28 @@ export function onOrganizationUpdate(
 
   // Update algolia's index
   if (before.name !== after.name) {
-    return storeSearchableOrg(orgID, after.name);
+    throw new Error('Organization name cannot be changed !'); // this will require to change the org ENS name, for now we throw to prevent silent bug
+    // const orgID = context.params.orgID;
+    // await storeSearchableOrg(orgID, after.name); // update org name on algolia
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/orgs.ts]: Remove  comments, or add todo with issue.

File: apps/backend-functions/src/orgs.ts [hunk 5]
```diff
@@ -43,7 +49,28 @@ export function onOrganizationUpdate(
...
+  }
+
+  // console.log(before, '****************************' ,after);
+  // Deploy org's smart-contract
+  if (before.status === OrganizationStatus.pending && after.status === OrganizationStatus.accepted) {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/orgs.ts]: Create a constant that explain this boolean value : 
```typescript
const becomeAccepted = before.status === OrganizationStatus.pending && after.status === OrganizationStatus.accepted;
if (becomeAccepted) ...
```

File: apps/backend-functions/src/orgs.ts [hunk 6]
```diff
@@ -43,7 +49,28 @@ export function onOrganizationUpdate(
...
+    const admin = await db.collection('users').doc(userIds[0]).get().then(adminSnapShot => adminSnapShot.data()!); // TODO use laurent's code after the merge of PR #698
+
+    const adminENS = emailToEnsDomain(admin.email, RELAYER_CONFIG.baseEnsDomain);
+    const orgENS = emailToEnsDomain(before.name.replace(' ', '-'), RELAYER_CONFIG.baseEnsDomain);
+    const isOrgRegistred = await isENSNameRegistred(adminENS, RELAYER_CONFIG);
+    if (isOrgRegistred) {
+      throw new Error(`This organization has already an ENS name : ${adminENS}`);
+    }
+    const adminAddress = await precomputeAddress(emailToEnsDomain(admin.email, RELAYER_CONFIG.baseEnsDomain), RELAYER_CONFIG);
+    const orgAddress =  await relayerDeployOrganizationLogic(adminAddress, RELAYER_CONFIG); // TODO promise all
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/orgs.ts]: Do promise.all

File: apps/backend-functions/src/relayer.ts [hunk 7]
```diff
@@ -281,3 +314,30 @@ export async function relayerSendLogic(
...
+    throw new Error('"adminAddress" is a mandatory parameters !');
+  }
+
+  try {
+    utils.getAddress(adminAddress);
+  } catch (error) {
+    throw new Error('"adminAddress" should be a valid ethereum address !');
+  }
+
+  const orgFactory = new ContractFactory(ORG_CONTRACT.abi, ORG_CONTRACT.bytecode, relayer.wallet);
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/relayer.ts]: call it `organizationFactory`

File: apps/backend-functions/src/utils.ts [hunk 8]
```diff
@@ -29,6 +29,12 @@ export function onDocumentUpdate(docPath: string, fn: Function) {
   .onUpdate(backup.skipWhenRestoring(fn));
 }
 
+export function onOrgDocumentUpdate(docPath: string, fn: Function) {
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/utils.ts]: org => organization

File: apps/backend-functions/src/internals/algolia.ts (hunk 1 above)
Comment (pldespaigne) [reply apps/backend-functions/src/internals/algolia.ts]: @lsenta is there a reason, or should I change this ?

Comment (lsenta) [reply apps/backend-functions/src/internals/algolia.ts]: Both are equivalent, stick to `Promise<any>` anyway, algolia returns a Task object and we won't react to this result.

**Pull request merged**